libs/ledger-live-common/src/families/internet_computer/bridge/account.ts

```diff
@@ -0,0 +1,262 @@
+import { makeAccountBridgeReceive, makeSync } from "../../../bridge/jsHelpers";
+import {
+  Account,
+  AccountBridge,
+  AccountLike,
+  BroadcastFnSignature,
+  Operation,
+  SignOperationEvent,
+  SignOperationFnSignature,
+} from "@ledgerhq/types-live";
+import { Transaction, TransactionStatus } from "../types";
+import { getAccountShape } from "./utils/account";
+import BigNumber from "bignumber.js";
+import { getEstimatedFees } from "./utils/token";
+import { getAddress, validateAddress, validateMemo } from "./utils/addresses";
+import {
+  AmountRequired,
+  InvalidAddress,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+  NotEnoughBalance,
+  RecipientRequired,
+} from "@ledgerhq/errors";
+import { Observable } from "rxjs";
+import { withDevice } from "../../../hw/deviceAccess";
+import { close } from "../../../hw";
+import { encodeOperationId } from "../../../operation";
+import {
+  broadcastTxn,
+  getTxnExpirationDate,
+  getTxnMetadata,
+  getUnsignedTransaction,
+  signICPTransaction,
+} from "./utils/icp";
+import { getPath } from "../utils";
+import { InvalidMemoICP } from "../errors";
+
+const sync = makeSync({ getAccountShape });
+
+const createTransaction = (): Transaction => {
+  // log("debug", "[createTransaction] creating base tx");
+  return {
+    family: "internet_computer",
+    amount: new BigNumber(0),
+    fees: getEstimatedFees(),
+    recipient: "",
+    useAllAmount: false,
+  };
+};
+
+const updateTransaction = (t: Transaction, patch: Transaction): Transaction => {
+  // log("debug", "[updateTransaction] patching tx");
+
+  return { ...t, ...patch };
+};
+
+const prepareTransaction = async (
+  a: Account,
+  t: Transaction
+): Promise<Transaction> => {
+  // log("debug", "[prepareTransaction] start fn");
+
+  const { address } = getAddress(a);
+  const { recipient } = t;
+
+  if (recipient && address) {
+    // log("debug", "[prepareTransaction] fetching estimated fees");
+
+    if (
+      (await validateAddress(recipient)).isValid &&
+      (await validateAddress(address)).isValid
+    ) {
+      if (t.useAllAmount) {
+        t.amount = a.spendableBalance.minus(t.fees);
+      }
+    }
+  }
+
+  // log("debug", "[prepareTransaction] finish fn");
+  return t;
```

Comment:
It's a bit of a hidden rule, we should document it better, but `t` here should not be mutated, otherwise it's not handled correctly on UI side.
The correct pattern is to either return `t` unchanged, as you do here, or return a new transaction object from `t` plus any new value: `return {...t, x, y, z}`

Reply:
Fixed